File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack = []
    for i in range(len(s)) :
        if s[i] == ')' :
            if stack :
                stack.pop()
            else :
                return False
        else :
            stack.append('(')
    if len(stack) > 0 :
        return False
    return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    Stack<Character> stack = new Stack<>();  
    
    boolean solution(String s) {
        boolean answer = true;
        for (int i=0; i < s.length(); i++) {
            if(s.charAt(i) == '(') {
                stack.push(s.charAt(i));
            } else {
                if (stack.size() > 0) {
                    stack.pop();
                } else {
                    return false;
                }
            }
        }
        if (stack.size() > 0) {
            return false;
        }
        return answer;
    }
}